las_geoh5/export_las.py

```diff
@@ -0,0 +1,205 @@
+    file.well.append(HeaderItem(mnemonic="X", value=float(drillhole.collar["x"])))
+    file.well.append(HeaderItem(mnemonic="Y", value=float(drillhole.collar["y"])))
+    file.well.append(HeaderItem(mnemonic="ELEV", value=float(drillhole.collar["z"])))
+
+    return file
+
+
+def add_curve_data(file: LASFile, drillhole: Drillhole, group):
+    """
+    Populate las file with curve data from each property in group.
+
+    :param file: lasio file object.
+    :param drillhole: geoh5py.drillhole object containing property
+        groups for collocated data.
+    :param group: Property group containing collocated float data
+        objects of 'drillhole'.
+    """
+
+    if not isinstance(group, ConcatenatedPropertyGroup):
+        raise TypeError("Property group must be of type ConcatenatedPropertyGroup.")
+
+    if group.depth_:
+        file.append_curve("DEPTH", group.depth_.values, unit="m")
+    else:
+        file.append_curve("DEPTH", group.from_.values, unit="m", descr="FROM")
+        file.append_curve("TO", group.to_.values, unit="m", descr="TO")
+
+    properties = [drillhole.get_data(k)[0] for k in group.properties]
+    for data in [
+        k for k in properties if k.name not in ["FROM", "TO", "DEPTH", "DEPT"]
+    ]:
+        file.append_curve(data.name, data.values)
+        if isinstance(data, ReferencedData):
+            for k, v in data.value_map.map.items():  # pylint: disable=invalid-name
+                file.params.append(
+                    HeaderItem(
+                        mnemonic=f"{data.name} ({k})", value=v, descr="REFERENCE"
+                    )
+                )
+
+    return file
+
+
+def add_survey_data(file: LASFile, drillhole: Drillhole) -> LASFile:
+    """
+    Add drillhole survey data to LASFile object.
+
+    :param file: las file object.
+    :param drillhole: drillhole containing survey data.
+
+    :return: Updated las file object.
+    """
+
+    # Add survey data
+    file.append_curve("DEPTH", drillhole.surveys[:, 0], unit="m")
+    file.append_curve(
+        "DIP",
+        drillhole.surveys[:, 1],
+        unit="degrees",
+        descr="from horizontal",
+    )
+    file.append_curve(
+        "AZIM",
+        drillhole.surveys[:, 2],
+        unit="degrees",
+        descr="from north (clockwise)",
+    )
+
+    return file
+
+
+def write_curves(
+    drillhole: Drillhole,
+    basepath: str | Path,
+    directory: bool = True,
+):
+    """
+    Write a formatted .las file for each property group in 'drillhole'.
+
+    :param drillhole: geoh5py drillhole object containing property
+        groups for collocated data.
+    :param basepath: Path to working directory.
+    :param directory: True if data is stored in sub-directories
+    """
+
+    if isinstance(basepath, str):
+        basepath = Path(basepath)
+
+    if not drillhole.property_groups:
+        raise AttributeError("Drillhole doesn't have any associated property groups.")
+
+    for group in drillhole.property_groups:

```

Comment:
Need to skip if `Azimuth/Dip` group, otherwise we get a crash `add_curve_data`

Reply:
Done.